python/ray/data/datasource/datasource.py

```diff
@@ -52,22 +52,17 @@ def prepare_read(self, parallelism: int, **read_args) -> List["ReadTask[T]"]:
 
     def do_write(
```

Comment:
Is it possible to make this change backwards compatible, like we do for the reader API?

Reply:
Added a new method for direct writing out, and deprecated `do_write()`.